### PR TITLE
SAK-48072 - Lessons: BltiPicker not working in Trinity

### PR DESCRIPTION
**Pull request by hornersa**
In BltiPicker.html the JS object “modal” is not part of the DOM which is why addEventListener cannot be invoked from that object. Use of the ‘el’ object instead works.

Besides some CSS styling for the LTI iframe presented in the modal, I added some "data-" attributes to force the backdrop of the modal to be static-- without which it’s frustratingly easy to inadvertently close the modal while attempting to fill out multiple data entry fields.

### CODE REVIEW

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm [hunk 1]
```diff
@@ -109,8 +109,9 @@ ${includeLatestJQuery}
...
 	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $formattedText.escapeHtml($alertMessage)</div>#end
         #if ($configMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $formattedText.escapeHtml($configMessage)</div>#end
-	#if ($isAdmin || $allowMaintainerAddSystemTool)<div align="right"><a href="" title="$tlang.getString("add.to.system")" onclick="location = '$sakai_ActionURL.setPanel("ToolInsert")';return false;"> $tlang.getString("add.to.system")</a>#if ($isAdmin )<br/><a href="" title="$tlang.getString("add.auto")" onclick="location = '$sakai_ActionURL.setPanel("AutoInsert")';return false;"> $tlang.getString("add.auto")</a>#end</div>#end
+	#if ($isAdmin || $allowMaintainerAddSystemTool)<div align="right" style="padding:10px"><a href="" title="$tlang.getString("add.to.system")" onclick="location = '$sakai_ActionURL.setPanel("ToolInsert")';return false;"> $tlang.getString("add.to.system")</a>#if ($isAdmin )<br/><a href="" title="$tlang.getString("add.auto")" onclick="location = '$sakai_ActionURL.setPanel("AutoInsert")';return false;"> $tlang.getString("add.auto")</a>#end</div>#end
 	<br/>
+        <div style="padding: 10px">
```
Comment (adrianfish) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm:114]: You can remove the inline styling and use a bs5 class. p-1, perhaps.

https://getbootstrap.com/docs/5.0/utilities/spacing/

File: lessonbuilder/tool/src/webapp/templates/BltiPicker.html [hunk 2]
```diff
@@ -131,22 +131,21 @@ <h5 class="modal-title" id="bltiModalLabel">Modal title</h5>
...
-      modal.addEventListener("hidden.bs.modal", e => {
 
+      el.addEventListener("hidden.bs.modal", e => {
         if ( doreload ) {
           location.reload();
         } else {
           $('#sakai-basiclti-admin-iframe').attr('src','/library/image/sakai/spinner.gif');
         }
-      });
+      });
```
Comment (adrianfish) [inline lessonbuilder/tool/src/webapp/templates/BltiPicker.html:148]:
```suggestion
      });
```

**Review by adrianfish — APPROVED** [review]
Some really minor stuff. Well, the inline style, that's pretty heavy. If I tell the other Shawn about that, he might have a word with you :)

Thanks for the fix.

**Comment by csev** [comment]
Thanks for finding and fixing this.

So I stumbled into this as well and filed another JIRA - https://sakaiproject.atlassian.net/browse/SAK-48093 - what is missing from this fix is the need to make these dialogs responsive.  In 22 and earlier, these were very carefully made be resized.  In the Trinity quick fix to this code, all the responsiveness went away.  In this patch, you hard-code the width a lot wider - which makes it look better - but I don't think it is responsive.   My patch builds on this patch and ends up with a responsive version of this.   I suggest we merge this patch.  I will test if it is responsive and then rebase my work on top of this patch to make the dialog responsive.

**Comment by adrianfish** [comment]
the bootstrap modals are responsive, to a degree. There are three sizes and they adjust to the breakpoints, without us needing any css. So, I suppose we just need to make sure we add our bits to the ootb bootstrap behaviour and not break the sizing that bootstrap tries to do, especially at small screen sizes.

**Comment by csev** [comment]
@adrianfish - After this gets merged I will test again.   In my testing, once shown, the modal started very narrow - about 1/3 of the width and then it did not change as I resized.  It started in what looked like the small breakpoint and then never changed.  But I cold be mistaken.

**Comment by csev** [comment]
I tested this PR and it works as you say.

File: basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm (hunk 1 above)
Comment (adrianfish) [inline basiclti/basiclti-tool/src/webapp/vm/lti_tool_system.vm:114]:
```suggestion
        <div class="p-1">
```

**Comment by hornersa** [comment]
Points noted regarding styling, breakpoints, etc. 

Hopefully with the central issue being resolved and progress toward resolving SAK-48090 underway, I can assess whether another Lessons/LTI issue is affecting Sakai 24/23 like it does for 21 (maybe 22). That is, the icon one chooses (now impacted by SAK-48090) when linking to an external tool is not actually asserted when the external tool link is instantiated on a Lessons page. Instead the default (a globe) is displayed regardless of the previous icon selection.

Once the aforementioned fixes are both on nightly (and after I try to complete some work on a rubrics issue) I intend to create a jira for this other Lessons/LTI icon bug, unless one already exists.

**Pull request merged**